guides/keras_cv/3d_object_detection_with_keras_cv.py

```diff
@@ -0,0 +1,367 @@
+  "point_mask": BooleanTensor[batch_size]
+}
+```
+
+The `point_xyz` field represents the XYZ coordinates of each point in the
+point cloud.
+
+The `point_features` field represents the LIDAR features of each point in the
+poin cloud. Typical features include range, intensity, and elongation.
+
+In KerasCV, 3D box targets for object detection are represented as vertical
+pillars rotated with respect to the Z axis. We encode each box as a list (or
+Tensor) of 7 floats: the X, Y, and Z coordinates of the box's center, the width,
+length, and height of the box, and the rotation of the box with respect to the
+Z axis. (This rotation is referrred to as `phi` and is always in radians).
+
+KerasCV's first 3D object detection model offering is a center-based model like
+the one proposed in https://arxiv.org/pdf/2006.11275.pdf.
+
+Let's get to 3D modelling!
+We'll start by loading up the Waymo Open Dataset. KerasCV provides a
+`waymo_open_dataset.load` function to load the Waymo Open Dataset into our
+data format.
+"""
+
+from keras_cv.datasets.waymo import load
+from keras_cv.datasets.waymo import build_tensors_for_augmentation
+
+# TODO(ianstenbit): Because we can't distribute WOD, we'll need to load wod_one_frame.tfrecord here
+data_shard = './training-data'
+dataset = load(data_shard)
+
+# By default, WOD point clouds are globally positioned, but for object detection
+# we want them with respect to the vehicle, so we transform them to the vehicle
+# frame of reference.
+dataset = dataset.map(transformer.transform_to_vehicle_frame)
+
+# Because number of points is dynamic, we pad them to make our inputs batchable.
+dataset = dataset.map(transformer.pad_or_trim_tensors)
+
+# Then we can easily reformat the tensors into KerasCV's data format!
+dataset = dataset.map(build_tensors_for_augmentation)
+
+# We use a small batch size here on CPU. Generally, point clouds can be pretty
+# large, so batch sizes are often smaller than in the 2D object detection world.
+dataset = dataset.batch(1)
+
+"""
+Loading up the Waymo Open Dataset can be a bit tricky, but this makes it pretty
+simple!
+
+One important note: Waymo Open Dataset is distributed as TFRecords representing
+a Waymo Open Dataset `Frame` proto. This cannot be deserialized in to Tensors
+inside of the TensorFlow graph, so this can cause CPU throttling during
+training.
+
+Therefore, KerasCV offers a utility for transforming Waymo Open Dataset frames
+into tf.Example records which can be more efficiently loaded into a TF graph
+for later training. The utility can be found at
+https://github.com/keras-team/keras-cv/blob/master/examples/training/object_detection_3d/waymo/serialize_records.py
+
+Next up, let's augment our data! In partnership with Waymo, KerasCV offers a
+set of state-of-the-art 3D augmentations for LIDAR data and 3D boxes. They
+behave like all Keras preprocessing layers, and they're very easy to set up.
+"""
+
+augmentations = keras.Sequential([
+  keras_cv.layers.GlobalRandomFlip(),
```

Comment:
lets explicitly provide the axis argument here `y=True` even if its the default.  Its a lot less obvious  what the sensible default is than in 2D